benchmarks/vector_db_compare/minima_treedb_runner.py

```
@@ -708,6 +715,51 @@ def _public_count_snapshot(self) -> dict[str, Any]:
+
+    def _batch_correlation_contract(self) -> dict[str, Any]:
+        if len(self.batch_correlations) > self._batch_correlation_max_records:
+            raise RuntimeError("TreeDB batch correlation count exceeds the frozen manifest bound")
+        compact_records = 0
+        full_records = 0
+        for correlation in self.batch_correlations:
+            if correlation.get("stats_retention") == "compact_completed":
+                if correlation.get("outcome") != "completed" or \
+                        correlation.get("profile_capture", {}).get("status") != "not_triggered" or \
+                        "before_stats" in correlation or "after_stats" in correlation:
+                    raise RuntimeError("compact TreeDB batch correlation retained diagnostic evidence")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)